src/python/pants/build_graph/build_configuration.py

```diff
@@ -179,11 +182,9 @@ def rules(self):
         """
         return list(self._rules)
 
-    def union_rules(self):
-        """Returns a mapping of registered union base types -> [OrderedSet of union member types].
-
-        :rtype: OrderedDict
-        """
+    def union_rules(self) -> typing.OrderedDict[Type, OrderedSet[Type]]:
```

Comment:
Hm, I would have expected there to be more type annotations added here, specifically to check that what we're putting into untyped `OrderedSet`s is the type we expect. `register_rules()` would be an example of a method I think could benefit from type annotations -- do you agree?

Reply:
It's certainly possible that there's some additional constraints we could put on `Type` here to make this signature more accurate - @Eric-Arellano ?